independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanProcessor.java

```diff
@@ -192,6 +192,8 @@ public List<Resource> generateResources(ReflectionRegistration reflectionRegistr
             ExecutorService executor)
             throws IOException, InterruptedException, ExecutionException {
 
+        beanDeployment.markAsResourcesGenerated();
```

Comment:
If it's called here the method should be named `resourceGenerationStarted()` or something like that. But that's only nitpicking ;-).

Reply:
@Ladicek I let you decide if you want to act on it. I'm fine with you merging and adjusting later (or not) if you prefer and want to avoid further conflicts.

Reply:
I thought about this (= `InvokerFactoryBuildItem` lifecycle) again and couldn't find a better solution. So I just applied @mkouba's suggestion.